bigquery/storage/managedwriter/options.go

```diff
@@ -278,8 +320,37 @@ type AppendOption func(*pendingWrite)
 // with a given stream.
 func UpdateSchemaDescriptor(schema *descriptorpb.DescriptorProto) AppendOption {
 	return func(pw *pendingWrite) {
-		// create a new descriptorVersion and attach it to the pending write.
-		pw.descVersion = newDescriptorVersion(schema)
+		prev := pw.reqTmpl
+		if prev == nil {
+			prev = newVersionedTemplate()
+		}
+		pw.reqTmpl = prev.revise(reviseProtoSchema(schema))
+	}
+}
+
+// UpdateMissingValueInterpretations updates the per-column missing-value intepretations settings,
+// and is retained for subsequent writes.  See the WithMissingValueInterpretations WriterOption for
+// more details.
+func UpdateMissingValueInterpretations(mvi map[string]storagepb.AppendRowsRequest_MissingValueInterpretation) AppendOption {
+	return func(pw *pendingWrite) {
+		prev := pw.reqTmpl
+		if prev == nil {
+			prev = newVersionedTemplate()
+		}
+		pw.reqTmpl = prev.revise(reviseMissingValueInterpretations(mvi))
```

Comment:
Why does this require two calls? Why doesn't `newVersionedTemplate` accept `...templateRevisionF` like `revise`?

Reply:
changed it to allowed revise() to recover from a nil start state, which simplifies the options.  We don't know based on the call order if we need to instantiate, which is why passing them into the newVersionedTemplate constructor doesn't get us as much.